STSampling.C

```diff
@@ -0,0 +1,565 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2019, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+#include "Matrix.h"
+#include "mpi.h"
+#include <cmath>
+#include <map>
+#include <set>
+
+#include "STSampling.h"
+#include "DEIM.h"
+
+using namespace std;
+
+namespace CAROM {
+
+// This function implements Algorithm 2 of Choi and Carlberg 2019.
+// All spatial indices are used here.
+// Spatial basis s_basis is distributed, of size (number of spatial DOFs) x (number of basis vectors)
+// Temporal basis t_basis is stored in its entirety on each process, of size (number of temporal DOFs) x (number of basis vectors)
+// In general, there may be multiple temporal basis vectors associated with each spatial basis vector, but here we assume for now
+// that there is only one, i.e. column j of s_basis corresponds to column j of t_basis.
+void SampleTemporalIndices(const Matrix* s_basis,
+                           const Matrix* t_basis,
+                           const int num_f_basis_vectors_used,
+                           int* t_samples,
+                           const int myid,
+                           const int num_procs,
+                           const int num_samples_req,
+                           const bool excludeFinalTime)
+{
+    CAROM_VERIFY(t_basis->distributed());
+
+    // Get the number of basis vectors and the size of each basis vector.
+    CAROM_VERIFY(0 < num_f_basis_vectors_used && num_f_basis_vectors_used <= s_basis->numColumns()
+                 && num_f_basis_vectors_used <= t_basis->numColumns());
+    CAROM_VERIFY(num_samples_req > 0);  // TODO: just replace num_samples with num_samples_req
+    const int num_basis_vectors =
+        std::min(num_f_basis_vectors_used, s_basis->numColumns());
+    //const int num_samples = num_samples_req > 0 ? num_samples_req : num_basis_vectors;
```

Comment:
Do we still need these commented out lines?

Reply:
Removed.